core/trino-main/src/main/java/io/trino/sql/planner/planprinter/DistributedPlanRepresentation.java

```diff
@@ -0,0 +1,564 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.planner.planprinter;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.airlift.units.DataSize;
+import io.airlift.units.Duration;
+import io.trino.cost.PlanCostEstimate;
+import io.trino.cost.PlanNodeStatsAndCostSummary;
+import io.trino.cost.PlanNodeStatsEstimate;
+import io.trino.sql.planner.Symbol;
+import io.trino.sql.planner.planprinter.NodeRepresentation.TypedSymbol;
+
+import javax.annotation.concurrent.Immutable;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static io.trino.sql.planner.planprinter.RendererUtils.formatAsCpuCost;
+import static io.trino.sql.planner.planprinter.RendererUtils.formatAsDataSize;
+import static io.trino.sql.planner.planprinter.RendererUtils.formatAsLong;
+import static io.trino.sql.planner.planprinter.RendererUtils.translateOperatorTypes;
+import static java.util.Collections.emptyMap;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toList;
+
+public class DistributedPlanRepresentation
+{
+    private final String name;
+    private final String identifier;
+    private final List<TypedSymbol> outputs;
+    private final String details;
+    private final DistributedPlanStats distributedPlanStats;
+    private final Optional<PlanNodeStatsAndCostSummary> planNodeStatsAndCostSummary;
+    private final List<EstimateStats> estimateStats;
+    private final List<DistributedPlanRepresentation> children;
+    private final Optional<JsonWindowOperatorStats> windowOperatorStats;
+
+    @JsonCreator
+    public DistributedPlanRepresentation(
+            @JsonProperty("name") String name,
+            @JsonProperty("identifier") String identifier,
+            @JsonProperty("layout") List<TypedSymbol> outputs,
+            @JsonProperty("details") String details,
+            @JsonProperty("reOrderJoinStatsAndCosts") Optional<PlanNodeStatsAndCostSummary> planNodeStatsAndCostSummary,
+            @JsonProperty("estimates") List<EstimateStats> estimateStats,
+            @JsonProperty("planStats") DistributedPlanStats distributedPlanStats,
+            @JsonProperty("children") List<DistributedPlanRepresentation> children,
+            @JsonProperty("windowOperatorStats") Optional<JsonWindowOperatorStats> windowOperatorStats)
```

Comment:
This should not be in this class. Any stats associated with window functions should be generated directly by the window operator